// react-native-share/lib/module/types.js
export let Social;

(function (Social) {
  Social["Facebook"] = "facebook";
  Social["FacebookStories"] = "facebookstories";
  Social["Pagesmanager"] = "pagesmanager";
  Social["Twitter"] = "twitter";
  Social["Whatsapp"] = "whatsapp";
  Social["Whatsappbusiness"] = "whatsappbusiness";
  Social["Instagram"] = "instagram";
  Social["InstagramStories"] = "instagramstories";
  Social["Googleplus"] = "googleplus";
  Social["Email"] = "email";
  Social["Pinterest"] = "pinterest";
  Social["Linkedin"] = "linkedin";
  Social["Sms"] = "sms";
  Social["Telegram"] = "telegram";
  Social["Snapchat"] = "snapchat";
  Social["Messenger"] = "messenger";
})(Social || (Social = {}));

export let ShareAsset;

(function (ShareAsset) {
  ShareAsset["BackgroundImage"] = "shareBackgroundImage";
  ShareAsset["BackgroundVideo"] = "shareBackgroundVideo";
  ShareAsset["StickerImage"] = "shareStickerImage";
  ShareAsset["BackgroundAndStickerImage"] = "shareBackgroundAndStickerImage";
})(ShareAsset || (ShareAsset = {}));
