// react-native-share/lib/commonjs/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ShareAsset = exports.Social = void 0;
let Social;
exports.Social = Social;

(function (Social) {
  Social["Facebook"] = "facebook";
  Social["FacebookStories"] = "facebookstories";
  Social["Pagesmanager"] = "pagesmanager";
  Social["Twitter"] = "twitter";
  Social["Whatsapp"] = "whatsapp";
  Social["Whatsappbusiness"] = "whatsappbusiness";
  Social["Instagram"] = "instagram";
  Social["InstagramStories"] = "instagramstories";
  Social["Googleplus"] = "googleplus";
  Social["Email"] = "email";
  Social["Pinterest"] = "pinterest";
  Social["Linkedin"] = "linkedin";
  Social["Sms"] = "sms";
  Social["Telegram"] = "telegram";
  Social["Snapchat"] = "snapchat";
  Social["Messenger"] = "messenger";
})(Social || (exports.Social = Social = {}));

let ShareAsset;
exports.ShareAsset = ShareAsset;

(function (ShareAsset) {
  ShareAsset["BackgroundImage"] = "shareBackgroundImage";
  ShareAsset["BackgroundVideo"] = "shareBackgroundVideo";
  ShareAsset["StickerImage"] = "shareStickerImage";
  ShareAsset["BackgroundAndStickerImage"] = "shareBackgroundAndStickerImage";
})(ShareAsset || (exports.ShareAsset = ShareAsset = {}));
